runelite-client/src/main/java/net/runelite/client/config/ConfigManager.java

```diff
@@ -331,7 +335,7 @@ private void saveToFile(final File propertiesFile) throws IOException
 
 		parent.mkdirs();
 
-		File tempFile = new File(parent, SETTINGS_FILE_NAME + ".tmp");
+		File tempFile = new File(parent, RuneLite.DEFAULT_CONFIG_FILE + ".tmp");
```

Comment:
This is probably not less-correct than before, but I think the config writing code still has races since it does an open+truncate and then lock. So, I think it could truncate a file while another RL is holding a lock and writing to it.